### ISSUE TITLE
Update custom exception codes

### ISSUE THREAD

**Issue by Silabs-ArjanB** [openhwgroup/cv32e40s#235]
Currently we use custom exception codes 48 (Instruction bus fault) and 49 (Instruction parity/checksum fault). It is not possible to use the etrigger functionality for these values. Consider using custom exception codes 24 and 25 instead.

48 -> 24 applies to the CV32E40X as well.

**Comment by Silabs-ArjanB** [comment]
Temporarily use LIB to differentiate old / new functionality (default to old functionality)

**Comment by Silabs-ArjanB** [comment]
Old exception codes enabled with LIB.7 == 0 && LIB.1 == 0:

48 (Instruction bus fault) 
49 (Instruction parity/checksum fault)

Old exception codes with LIB.7 == 1 && LIB.1 == 1:

24 (Instruction bus fault) 
25 (Instruction parity/checksum fault)

**Comment by Silabs-ArjanB** [comment]
Documentation has already been updated: https://github.com/openhwgroup/cv32e40s/pull/269, https://github.com/openhwgroup/cv32e40x/pull/627

**Comment by Silabs-ArjanB** [comment]
The instruction bus fault update needs to be done first on the 40X, then merged into the 40S
